File: ios/NoodleManager.m
//
//  NoodleManager.m
//  BridgeExample
//
//  Created by Julien Karst on 29/10/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "NoodleManager.h"
#import <React/RCTLog.h>

@implementation NoodleManager

// To export a module named NoodleManager
RCT_EXPORT_MODULE();

// Get property with name and trigger a callback to the javascript thread
RCT_EXPORT_METHOD(addNoodle:(NSString *)name response:(RCTResponseSenderBlock)callback)
{
  RCTLogInfo(@"We are cooking your Noodle %@ ", name);
  callback(@[name, @"Your noodle is now ready !"]);
}

@end

File: android/app/src/main/java/com/bridgeexample/NoodleManager.java
package com.bridgeexample;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;


public class NoodleManager extends ReactContextBaseJavaModule {
    private static ReactApplicationContext reactContext;

    NoodleManager(ReactApplicationContext context) {
        super(context);
        reactContext = context;
    }

    // Will be the name of your native module
    @Override
    public String getName() {
        return "NoodleManager";
    }

    @ReactMethod
    public void addNoodle(String name, Callback response) {
        response.invoke(name, "your noodle are ready !");
    }
}